uart_example/lib/main.dart

```diff
@@ -0,0 +1,342 @@
+      id:_foundBleUARTDevices[index].id,
+      prescanDuration: Duration(seconds: 1),
+      withServices: [_UART_UUID, _UART_RX, _UART_TX],
+    );
+    _logTexts = "";
+    refreshScreen();
+    _connection = _currentConnectionStream.listen((event) {
+      String id = event.deviceId.toString();
+      switch(event.connectionState) {
+        case DeviceConnectionState.connecting:
+          {
+            _logTexts = _logTexts + "Connecting to " + id +"\n";
+            break;
+          }
+        case DeviceConnectionState.connected:
+          {
+            _connected = true;
+            _logTexts = _logTexts + "Connected to " + id + "\n";
+            _numberOfMessagesReceived = 0;
+            _receivedData = [];
+            _txCharacteristic = QualifiedCharacteristic(serviceId: _UART_UUID, characteristicId: _UART_TX, deviceId: event.deviceId);
+            _receivedDataStream = flutterReactiveBle.subscribeToCharacteristic(_txCharacteristic);
+            _receivedDataStream.listen((data) {
+               onNewReceivedData(data);
+            }, onError: (dynamic error) {
+              _logTexts = _logTexts + "Error:" + error.toString() + id + "\n";
+            });
+            _rxCharacteristic = QualifiedCharacteristic(serviceId: _UART_UUID, characteristicId: _UART_RX, deviceId: event.deviceId);
+            break;
+          }
+        case DeviceConnectionState.disconnecting:
+          {
+            _connected = false;
+            _logTexts = _logTexts + "Disconnecting from " + id + "\n";
+            break;
+          }
+        case DeviceConnectionState.disconnected:
+          {
+            _logTexts = _logTexts + "Disconnected from " + id + "\n";
+            break;
+          }
+      }
+      refreshScreen();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        // Here we take the value from the MyHomePage object that was created by
+        // the App.build method, and use it to set our appbar title.
+        title: Text(widget.title),
+      ),
+      body: SingleChildScrollView(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: <Widget>[
+            Text("BLE UART Devices found:"),
+            Container(
+                margin: const EdgeInsets.all(3.0),
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(10),
+                border: Border.all(
+                  color: Colors.blue,
+                  width:2
+                )
+              ),
+              height: 100,
+              child: ListView.builder(
+                  itemCount: _foundBleUARTDevices.length,
+                  itemBuilder: (context, index) {
+                    return Card(
+                        child: ListTile(
+                          dense: true,
+                          enabled: !((!_connected && _scanning) || (!_scanning && _connected)),
```

Comment:
`!a && b || a && !b` is equivalent to `a != b` (exclusive OR) and `!(a != b)` is `a == b`, so this whole expression should be equivalent to `_scanning == _connected`, if I didn't miss anything.